server/src/main/java/org/elasticsearch/discovery/PeerFinder.java

```diff
@@ -380,6 +389,7 @@ private void requestPeers() {
 
             transportService.sendRequest(discoveryNode, REQUEST_PEERS_ACTION_NAME,
                 new PeersRequest(getLocalNode(), knownNodes),
+                TransportRequestOptions.builder().withTimeout(requestPeersTimeout).build(),
```

Comment:
Black-holing of traffic led to a liveness failure when a `PeersRequest` was dropped.